src/js/angular/guides/tour-lib-services/shepherd.service.js

```diff
@@ -0,0 +1,543 @@
+     *       canBePaused: boolean,
+     *       onNextClick: function,
+     *       onPreviousClick: function
+     *     }
+     * </pre>
+     *
+     * <ul>
+     *     <li>
+     *         <b>title</b> - the title of the step. It will be shown as header of popover dialog;
+     *     </li>
+     *     <li>
+     *         <b>content</b> - the content of the step. It will be shown as content of popover dialog;
+     *     </li>
+     *     <li>
+     *         <b>elementSelector</b> - the selector of the element where popover dialog will be placed.
+     *     </li>
+     *     <li>
+     *         <b>url</b> - the url of the page where the element is present;
+     *     </li>
+     *     <li>
+     *         <b>placement</b> - where popover dialog to be placed. Possible values:
+     *         <ul>
+     *           <li>auto</li>
+     *           <li>auto-start</li>
+     *           <li>auto-end</li>
+     *           <li>top</li>
+     *           <li>top-start</li>
+     *           <li>top-end</li>
+     *           <li>bottom</li>
+     *           <li>bottom-start</li>
+     *           <li>bottom-end</li>
+     *           <li>right</li>
+     *           <li>right-start</li>
+     *           <li>right-end</li>
+     *           <li>left</li>
+     *           <li>left-start</li>
+     *           <li>left-end</li>
+     *         </ul>
+     *     </li>
+     *     <li><b>type</b> - the type of step. Possible values:
+     *       <ul>
+     *           <li>
+     *               <b>clickable-element</b> - the element is clickable like link, button... Client of workbench can click on this element.
+     *           </li>
+     *           <li>
+     *               <b>read-only-element</b> - the element is readable. Client of workbench can only read this component.
+     *           </li>
+     *           <li>
+     *               <b>input-element</b> - the element is some kind ot input like text, text area ... . Client of workbench can enter data in the element.
+     *           </li>
+     *        </ul>
+     *     </li>
+     *     <li>
+     *         <b>maxWaitTime</b> - some element need more time to be present on the page. This is the max waiting time element to be visible
+     *         on the page. Default value is 3 second.
+     *     </li>
+     *     <li>
+     *         <b>canBePaused</b> - some element need user interaction with some other element on the page before appear. Step which uses such
+     *         element can be persist when guide is paused. For example if step is "Click on sub menu "Repositories" of menu "Setup""
+     *     </li>
+     *     <li>
+     *         <b>onNextClick</b> - function which will be executed when nex button is clicked.
+     *     </li>
+     *      <li>
+     *         <b>onPreviousClick</b> - function which will be executed when previous button is clicked.
+     *     </li>
+     * </ul>
+     */
+    this._addGuideSteps = (guide, stepsDescriptions) => {
```

Comment:
I don't get the logic in this function. Also, these `if` statements should be transformed to `if-else` IMO.

Reply:
The first if, processes only first step if length is greater than 0 this mean stepsDescriptions is not empty and we process first step. The first step is different than other that it has not previous step.
	The second if, processes only middle steps, if length is greater than 2 this mean that stepsDescriptions has middle steps. The middle steps are different that they have previous and next steps.
	The last if, processes only last step, if length is greater than 1 this mean that we have more than one step. The last step is different than other that it has not next step.
	I will add comments to every one if.